### ISSUE TITLE
Add publishing feature

### ISSUE THREAD

**Issue by imouaddine** [imouaddine/Stylejonction#9]
Use case:

User added some changes to his portfolio, those changes are not reflected immediately on his public page until he publish it.
However, the changes should be persistant and saved in the database. In other words, we will have two versions of a portfolio including its associations( projects, layout and etc..): draft and published version.

If it's too complicated to implement, let me know. 

I'm always available if you have any questions.

**Comment by ignacy** [comment]
I have some questions about this. 

We don't have a "public portfolio page" so we should probably create it first. Also as far as the users go public portfolio can be viewed by anyone (even non-logged users)? And when sending invitations to project, we're inviting other users (because we have to verify the email), does that mean that we can only invite other designers that are registered, or will we have 4 user types instead of 3:
- admin
- designer
- user with account but not a designer
- user without account

**Comment by imouaddine** [comment]
We should have only two types of users:
- admin 
- designer or owner portfolio

Regarding invitation, designer could invite anyone with adress email and no **email verification is required**. It's his responsibility to check the validity of the email. We want to keep it very simple. 

Invitees should be able to access the public page of a project in the portfolio with no registration required, only with the link sent in the invitation email. This link should expire after 48 hours.

**Comment by ignacy** [comment]
OK thank you for explaining this.

So I understand from the above that I will need to redirect invitation to a page like this:
stylejunction.com/projects?token=asdfasdfasdfasdfasdfasdfasdfas

**Comment by imouaddine** [comment]
Additional notes:
- Show action of portfolio controller will render the public portfolio
- Edit action of portfolio controller will redirect to edit_layout action
- For html format pblish action will publish the portfolio and redirect to the current page and for json format a notice message will be returned.

**Comment by imouaddine** [comment]
Please add publish button to the header of the portfolio editor to be able to validate that it works.

**Comment by imouaddine** [comment]
We will abdondon this feature. The current implementation is not working correctly and it will be hard to maintain. On my side, I tried another method using papertrail gem used for versioning, I was able to make it work after a day of work. But I  came to conclusion, we should abandon this feature completly because we're going to add another models and new association to the portfolio It will be too much hassle to maintain and to test.
Bottom line: I will convince the client that the implementation of the feature is too complicated and we will keep publishing very simple, similar to what have you implemented at the beginning. User can publish or unpublish the portfolio. if the portfolio is unpublished It will not be public and vice versa.

I apologize for the wasted effort.

I have created a new issue assigned to me to do the rollback because I need the changes asap.
